test/FunctionalTests/Microsoft.OData.Client.Tests/ALinq/DollarApplyGroupByTests.cs

```diff
@@ -0,0 +1,1717 @@
+                aggregateQuery.ToString()
+            );
+
+            MockGroupBy_Max_ByConstant();
+
+            var aggregateResult = aggregateQuery.ToArray();
+
+            Assert.Single(aggregateResult);
+
+            var singleResult = aggregateResult.First();
+
+            Assert.Equal(141, singleResult.MaxIntProp);
+            Assert.Equal(199, singleResult.MaxNullableIntProp);
+            Assert.Equal(155.85, singleResult.MaxDoubleProp);
+            Assert.Equal(178.49, singleResult.MaxNullableDoubleProp);
+            Assert.Equal(173.90M, singleResult.MaxDecimalProp);
+            Assert.Equal(157.30M, singleResult.MaxNullableDecimalProp);
+            Assert.Equal(300L, singleResult.MaxLongProp);
+            Assert.Equal(295L, singleResult.MaxNullableLongProp);
+            Assert.Equal(171.22f, singleResult.MaxSingleProp);
+            Assert.Equal(174.99f, singleResult.MaxNullableSingleProp);
+        }
+
+        [Fact]
+        public void GroupByResultSelector_Max_BySingleProperty()
+        {
+            var queryable = this.dsContext.CreateQuery<Number>(numbersEntitySetName);
+
+            var aggregateQuery = queryable.GroupBy(d1 => d1.RowParity, (d2, d3) => new
+            {
+                RowParity = d2,
+                MaxIntProp = d3.Max(d4 => d4.IntProp),
+                MaxNullableIntProp = d3.Max(d4 => d4.NullableIntProp),
+                MaxDoubleProp = d3.Max(d4 => d4.DoubleProp),
+                MaxNullableDoubleProp = d3.Max(d4 => d4.NullableDoubleProp),
+                MaxDecimalProp = d3.Max(d4 => d4.DecimalProp),
+                MaxNullableDecimalProp = d3.Max(d4 => d4.NullableDecimalProp),
+                MaxLongProp = d3.Max(d4 => d4.LongProp),
+                MaxNullableLongProp = d3.Max(d4 => d4.NullableLongProp),
+                MaxSingleProp = d3.Max(d4 => d4.SingleProp),
+                MaxNullableSingleProp = d3.Max(d4 => d4.NullableSingleProp),
+            });
+
+            Assert.Equal(
+                string.Format("{0}/Numbers?$apply=groupby((RowParity),aggregate(" +
+                "IntProp with max as MaxIntProp,NullableIntProp with max as MaxNullableIntProp," +
+                "DoubleProp with max as MaxDoubleProp,NullableDoubleProp with max as MaxNullableDoubleProp," +
+                "DecimalProp with max as MaxDecimalProp,NullableDecimalProp with max as MaxNullableDecimalProp," +
+                "LongProp with max as MaxLongProp,NullableLongProp with max as MaxNullableLongProp," +
+                "SingleProp with max as MaxSingleProp,NullableSingleProp with max as MaxNullableSingleProp))", serviceUri),
+                aggregateQuery.ToString()
+            );
+
+            MockGroupBy_Max_BySingleProperty();
+
+            var aggregateResult = aggregateQuery.ToArray();
+
+            Assert.Equal(2, aggregateResult.Length);
+            Assert.Equal("Even", aggregateResult[0].RowParity);

```

Comment:
nit: Maybe you could also assert some of the aggregated properties for at least one of the scenarios with a non-constant grouping expression just to be sure they don't disappear when you have a groupby clause :)

Reply:
Done for this and other tests